script.js

```diff
@@ -0,0 +1,156 @@
+      alert("이미 같은 할 일이 존재합니다!");
+      input.value = ""; //사용자 입력칸 빈칸 리셋
+      input.focus(); //입력창에 포커스
+    } else {
+      addToList(text, false);
+      saveLocalStorage(text, false);
+      input.value = ""; //사용자 입력칸 빈칸 리셋
+      input.focus(); //입력창에 포커스
+    }
+  } else {
+    alert("할 일을 작성해주세요!");
+  }
+
+  getDoneCount();
+  getTodoCount();
+});
+
+let doneCount = 0;
+
+function addToList(text, checked) {
+  const list = document.querySelector("#list");
+  let li = document.createElement("li");
+  let checkBox = document.createElement("input");
+  checkBox.type = "checkbox";
+  checkBox.checked = checked;
+
+  let span = document.createElement("span");
+  span.textContent = text;
+  //삭제 버튼 추가
+  let deleteBtn = document.createElement("button");
+  deleteBtn.textContent = "삭제";
+
+  //체크 여부에 따라 스타일 변경
+  if (checked) {
+    span.style.textDecoration = "line-through";
+    span.style.color = "gray";
+  }else{
+    span.style.color="white"
+  }
+
+  checkBox.addEventListener("change", () => {
+    doneList(text, checkBox.checked);
+
+    span.style.textDecoration = checkBox.checked ? "line-through" : "none";
+    span.style.color = checkBox.checked ? "gray" : "white";
+
+    getTodoCount();
+    getDoneCount();
+
+    let todos = JSON.parse(localStorage.getItem("todos")) || [];
+    if (doneCount === todos.length) {
+      alert("축하합니다! 모든 할 일을 다하셨어요!");
+    }
+  });
+
+  //삭제버튼 클릭 시 해당 아이템 삭제
+  deleteBtn.addEventListener("click", () => {
+    if (confirm("할 일을 삭제하시겠습니까?")) {
+      li.remove();
+      deleteLocalStorage(text);
+      getTodoCount();
+      getDoneCount();
+      alert("삭제 완료");
+    } else {
+      alert("삭제 취소");
+    }
+  });
+
+  li.append(checkBox);
+  li.append(span);
+  li.append(deleteBtn);
+  list.append(li); // 리스트에 li 추가
+}
+
+//로컬스토리지에 저장
+function saveLocalStorage(text, checked) {
+  let todos = JSON.parse(localStorage.getItem("todos")) || [];
+  todos.push({ text, checked });
+  localStorage.setItem("todos", JSON.stringify(todos));
+}
+
+//로컬스토리지 데이터 가져오기
+function loadList() {
+  let todos = JSON.parse(localStorage.getItem("todos")) || [];
+  todos.forEach(({ text, checked }) => addToList(text, checked));
+
+  getDoneCount();
+  getTodoCount();
+}
+
+function doneList(text, checked) {
+  let todos = JSON.parse(localStorage.getItem("todos")) || [];
+  todos = todos.map((todo) =>
+    todo.text === text ? { ...todo, checked: checked } : todo
+  );
+
+  localStorage.setItem("todos", JSON.stringify(todos));
+
+  getDoneCount();
+  getTodoCount();
+}
+
+function getTodoCount() {
+  let todos = JSON.parse(localStorage.getItem("todos")) || [];
+  let todoCount = todos.filter((todo) => !todo.checked).length;
+
+  document.querySelector("#todoCount").innerHTML = todoCount;
+}
+
+function getDoneCount() {
+  let todos = JSON.parse(localStorage.getItem("todos")) || [];
+  doneCount = todos.filter((todo) => todo.checked).length;
+
+  document.querySelector("#doneCount").innerHTML = doneCount;
+}
```

Comment:
사소하지만 개인적으로 ```let todos = JSON.parse(localStorage.getItem("todos")) || [];```가 중복으로 호출되고 있는데, 두 함수는 함께 쓰이고 있으니 하나로 합치고 호출도 한 번으로 줄이면 좋을 것 같아요!
```suggestion
function updateCounts() {
  let todos = JSON.parse(localStorage.getItem("todos")) || [];
  let todoCount = todos.filter((todo) => !todo.checked).length;
  let doneCount = todos.filter((todo) => todo.checked).length;

  document.querySelector("#todoCount").innerHTML = todoCount;
  document.querySelector("#doneCount").innerHTML = doneCount;
}
```